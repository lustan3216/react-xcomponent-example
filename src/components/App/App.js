import React from 'react'
import PropTypes from 'prop-types'
import Logo from '../Logo'
import Message from '../Message'
import styles from './App.css'

const App = ({ message }) => (
  <div className={styles.container}>
    <Logo />
    <Message text={message} />
  </div>
)

const { string } = PropTypes
App.propTypes = {
  message: string.isRequired
}

export default App
