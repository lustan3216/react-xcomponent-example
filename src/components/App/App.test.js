import React from 'react'
import App from './'
import renderer from 'react-test-renderer'

test('App snapshot', () => {
  const component = renderer.create(<App message="It Works!" />)
  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})
