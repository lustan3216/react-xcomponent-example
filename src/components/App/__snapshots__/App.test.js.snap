// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div
  className="container"
>
  <img
    src="test-file-stub"
  />
  <span
    className="message"
  >
    It Works!
  </span>
</div>
`;
